test/e2e/tests/test_adopted_resource.py

```diff
@@ -0,0 +1,201 @@
+        assert deleted
+
+
+@pytest.fixture(scope="module")
+def adopt_scaling_policy(adopt_scalable_target):
+    resource_id, adopted_target_reference = adopt_scalable_target
+    policy_resource_name = random_suffix_name("sagemaker-scaling-policy", 32)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["ADOPTED_POLICY_NAME"] = policy_resource_name
+    replacements["RESOURCE_ID"] = resource_id
+
+    adopted_policy_reference, _, adopted_policy_resource = create_adopted_resource(
+        resource_name=policy_resource_name,
+        spec_file="sagemaker_endpoint_adopted_policy",
+        replacements=replacements,
+    )
+
+    assert adopted_policy_resource is not None
+
+    yield (resource_id, adopted_target_reference, adopted_policy_reference)
+
+    if k8s.get_resource_exists(adopted_policy_reference):
+        _, deleted = k8s.delete_custom_resource(adopted_policy_reference)
+        assert deleted
+
+
+@service_marker
+@pytest.mark.canary
+class TestAdoptedSageMakerEndpointAutoscaling:
+    def test_smoke(self, put_scaling_policy, adopt_scaling_policy):
+        sdk_resource_id = put_scaling_policy
+
+        (
+            adopted_resource_id,
+            adopted_target_reference,
+            adopted_policy_reference,
+        ) = adopt_scaling_policy
+
+        assert sdk_resource_id == adopted_resource_id
+
+        target_name = k8s.get_resource(adopted_target_reference)["spec"]["kubernetes"][
+            "metadata"
+        ]["name"]
+        policy_name = k8s.get_resource(adopted_policy_reference)["spec"]["kubernetes"][
+            "metadata"
+        ]["name"]
+
+        assert target_name is not None
+        assert policy_name is not None
+
+        for reference in (
+            adopted_target_reference,
+            adopted_policy_reference,
+        ):
+            assert k8s.wait_on_condition(reference, "ACK.Adopted", "True")
+
+        target_reference = k8s.create_reference(
+            CRD_GROUP, CRD_VERSION, TARGET_RESOURCE_PLURAL, target_name, "default"
+        )
+        target_resource = k8s.wait_resource_consumed_by_controller(target_reference)
+        assert target_resource is not None
+
+        assert target_resource["spec"].get("resourceID", None) == sdk_resource_id
+
+        policy_reference = k8s.create_reference(
+            CRD_GROUP, CRD_VERSION, POLICY_RESOURCE_PLURAL, policy_name, "default"
+        )
+        policy_resource = k8s.wait_resource_consumed_by_controller(policy_reference)
+        assert policy_resource is not None
+
+        assert policy_resource["spec"].get("resourceID", None) == sdk_resource_id
+
+        # Delete the Adopted Resources
+        _, deleted = k8s.delete_custom_resource(adopted_policy_reference)
+        assert deleted is True
+
+        _, deleted = k8s.delete_custom_resource(adopted_target_reference)
+        assert deleted is True
+
+        # Delete the ApplicationAutoscaling Resources
```

Comment:
@surajkota This is what was missing in the previous PR which meant that boto3 deletion was being used. 

[reference](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/pull/42#discussion_r706532786)

Reply:
Are these necessary? We are testing for deletion in the fixtures (they delete and assert)

Reply:
yes, delete is getting tested but the intention is also to clean up the resources created by the test. if we don't have this, the actual resource will get deleted and the controller will error if it reconciles

Reply:
added delete logic.